src/integrations/feed/clients.ts

```diff
@@ -0,0 +1,167 @@
+import { FeedConfig, FeedResponse, IFeedClient } from './types';
+import fetch, { RequestInit } from 'node-fetch';
+import { fetchOptions as globalFetchOptions } from '../../http';
+import pRetry, { AbortError } from 'p-retry';
+import { IORedisPool } from '@dailydotdev/ts-ioredis-pool';
+import { Context } from '../../Context';
+
+const ONE_DAY_SECONDS = 24 * 60 * 60;
+
+/**
+ * Naive implementation of a feed client that fetches the feed from the feed service
+ */
+export class FeedClient implements IFeedClient {
+  private readonly url: string;
+  private readonly fetchOptions: RequestInit;
+
+  constructor(
+    url = process.env.INTERNAL_FEED,
+    fetchOptions: RequestInit = globalFetchOptions,
+  ) {
+    this.url = url;
+    this.fetchOptions = fetchOptions;
+  }
+
+  private async internalFetchFeed(config: FeedConfig): Promise<{
+    data: { post_id: string; metadata: Record<string, string> }[];
+  }> {
```

Comment:
Not really blocking, but maybe make this a interface return type? (Just more readable to me)

Reply:
Actually would `FeedResponse` not do?

Reply:
`FeedResponse` is a bit different. I'll create a dedicated type, no worries